test/functional/autocmd/cursorhold_spec.lua

```diff
@@ -0,0 +1,34 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local eq = helpers.eq
+local eval = helpers.eval
+local feed = helpers.feed
+local retry = helpers.retry
+local source = helpers.source
+
+describe('CursorHold', function()
+  before_each(clear)
+
+  it('is triggered after timer in InsertEnter', function()
+    source([[
+    set updatetime=1
+
+    function F()
+      if mode() ==# 'i'
+      endif
+    endfunction
+
+    let g:cursorhold = 0
+    augroup test
+      au CursorHoldI * let g:cursorhold += 1
+      au InsertEnter * call timer_start(0, {-> 1})
```

Comment:
The InsertEnter is not required to trigger the issue here actually, I assume since it uses events I assume.